### PR TITLE
feat: MMF 

### PR DESCRIPTION
**Pull request by croc12344**
## Description
Adds the pools from mm.finance

## Checklist

- [ ] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] (optional) As a contributor, my Ethereum address/ENS is:
- [ ] (optional) As a contributor, my Twitter handle is:

### CODE REVIEW

File: src/apps/mmfinance/cronos/MMfinance.auto-cake.contract-position-fetcher.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      resolveRewardTokenAddresses: ({ multicall, contract }) => multicall.wrap(contract).token(),
+      rewardRateUnit: RewardRateUnit.BLOCK,
+      resolveRewardRate: this.appToolkit.helpers.masterChefDefaultRewardsPerBlockStrategy.build({
+        resolvePoolAllocPoints: ({ multicall }) =>
+          multicall
+            .wrap(chefContract)
+            .poolInfo(0)
+            .then(i => i.allocPoint),
+        resolveTotalAllocPoints: ({ multicall }) => multicall.wrap(chefContract).totalAllocPoint(),
+        resolveTotalRewardRate: ({ multicall }) => multicall.wrap(chefContract).cakePerBlock(),
```
Comment (immasandwich) [inline src/apps/mmfinance/cronos/MMfinance.auto-cake.contract-position-fetcher.ts:50]: Each of these methods has access to the `contract` as well... i.e.:

```ts
resolveTotalRewardRate: ({ multicall, contract }) => multicall.wrap(contract).cakePerBlock(),
```

...so you won't need to declare `chefContract` on line 27.

File: src/apps/mmfinance/cronos/MMfinance.pool.cache-manager.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  //   return compact(allAddresses);
+  // }
+
+  @CacheOnInterval({
+    key: `apps-v3:${MMFINANCE_DEFINITION.id}:static-pool-addresses`,
+    timeout: 15 * 60 * 1000,
+  })
+  private async getStaticPoolAddresses() {
+    return [''];
+  }
```
Comment (immasandwich) [inline src/apps/mmfinance/cronos/MMfinance.pool.cache-manager.ts:127]: You can remove this method and the commented out method. Easy to add back in later.

File: src/apps/mmfinance/mmfinance.definition.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    telegram: 'https://t.me/mmfinance',
+    medium: 'https://mmfinance.medium.com/',
+  },
+  groups: {
+    pool: { id: 'pool', type: GroupType.TOKEN, label: 'Pools' },
+    // farmV2: { id: 'farm-v2', type: GroupType.POSITION, label: 'Farms' },
+    // syrupStaking: { id: 'syrup-staking', type: GroupType.POSITION, label: 'Syrup Pools' },
+    // syrupMmf: { id: 'syrup-mmf', type: GroupType.POSITION, label: 'Syrup Pools' },
+    // autoMmf: { id: 'auto-mmf', type: GroupType.POSITION, label: 'Auto MMF', isHiddenFromExplore: true }, // This is the old staking mmf staking contract, and the rewards are being counted towards the TVL
+    // ifoMmf: { id: 'ifo-mmf', type: GroupType.POSITION, label: 'IFO Mmf', isHiddenFromExplore: true },
```
Comment (immasandwich) [inline src/apps/mmfinance/mmfinance.definition.ts:24]: With these commented out, you'll have type errors in your files that are referencing these groups. Uncomment these line, and simply have the files commented out in the module like you have already done.

**Comment by immasandwich** [comment]
@croc12344 let me know when you're ready for another review, the PR is still marked as a draft.

File: src/apps/mmfinance/cronos/MMfinance.auto-mmf.contract-position-fetcher.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { Inject } from '@nestjs/common';
```
Comment (immasandwich) [inline src/apps/mmfinance/cronos/MMfinance.auto-mmf.contract-position-fetcher.ts:1]: Please lowercase the `mmfinance` in all of the filenames (we want it to match the name of the directory `mmfinance` for consistency)... i.e.: `src/apps/mmfinance/cronos/mmfinance.auto-mmf.contract-position-fetcher.ts`.

File: src/apps/mmfinance/cronos/MMfinance.auto-mmf.contract-position-fetcher.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(MmfinanceContractFactory)
+    private readonly contractFactory: MmfinanceContractFactory,
+  ) { }
+
+  getPositions() {
+    const chefContract = this.contractFactory.mmfinanceChef({
+      network,
+      address: '0x6bE34986Fdd1A91e4634eb6b9F8017439b7b5EDc',
```
Comment (immasandwich) [inline src/apps/mmfinance/cronos/MMfinance.auto-mmf.contract-position-fetcher.ts:29]: Keep all addresses lowercased. It helps with consistency with the rest of our integrations.

File: src/apps/mmfinance/cronos/MMfinance.auto-mmf.contract-position-fetcher.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+const groupId = MMFINANCE_DEFINITION.groups.autoMmf.id;
+const network = Network.CRONOS_MAINNET;
+
+@Register.ContractPositionFetcher({ appId, groupId, network })
+export class CronosChainMmfinanceAutoCakeContractPositionFetcher implements PositionFetcher<ContractPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(MmfinanceContractFactory)
+    private readonly contractFactory: MmfinanceContractFactory,
+  ) { }
```
Comment (immasandwich) [inline src/apps/mmfinance/cronos/MMfinance.auto-mmf.contract-position-fetcher.ts:24]: Please ensure your IDE is configured to use the linting settings in our repository. This should be `{}` not `{ }`. You can alternatively run the linting scripts in `package.json`.

**Comment by SamIam-0x** [comment]
Hello @croc12344 !

**Comment by SamIam-0x** [comment]
Gentle ping on the above @croc12344 ! Thanks!

**Pull request closed without merging**